### DISCUSSION TITLE
Should @implements have singleton=False?

### DISCUSSION THREAD

**Discussion by pauleveritt** [Finistere/antidote#61] [General]
Not reporting as a bug, as I'm not sure the intention.

I view `@implements` as kind of isomorphic with `@injectable`. Thus, you should be able to register non-singleton implementations.

But perhaps I'm looking at this wrong?

**Comment by Finistere** [comment]
It's indeed missing an example. `@implements` doesn't replace in any way `@injectable`, it only applies it when not done already.

```python
from antidote import interface, implements, injectable, world


@interface
class Greeter:
    def greet(self) -> str:
        raise NotImplementedError()


@implements(Greeter)
@injectable(singleton=False)
class FrenchGreeter(Greeter):
    def greet(self) -> str:
        return f"{self}: Bonjour!"

```

In the same spirit, `@lazy` relies on `@inject` in the same way and the auto-wiring of `@injectable` / `@wire` won't touch any existing injected methods.
